[pytorch][codegen] move is_abstract property to NativeFunction model

ljk53 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #48249 [pytorch][codegen] add autograd data model
* **#48252 [pytorch][codegen] move is_abstract property to NativeFunction model**

Moved to a shared place so that gen_variable_type.py can reuse it.

Differential Revision: [D25087808](https://our.internmc.facebook.com/intern/diff/D25087808)

ljk53 committed:
[pytorch][codegen] move is_abstract property to NativeFunction model

Moved to a shared place so that gen_variable_type.py can reuse it.

[ghstack-poisoned]

ezyang commented on tools/codegen/model.py:
```diff
@@ -125,6 +125,29 @@ class NativeFunction:
+    # whether or not there is an explicit entry in derivatives.yaml
+    # for the entry or not (as this affects whether or not the operation is
+    # overrideable or not.)  Once this all gets cleaned up, this
+    # property will be obsolete.
```
So.... is it or is it not useful for end users? :)

ljk53 replied:
I wasn't entirely sure whether the original comment was talking about external users of Declarations.yaml (it's kinda useful for autograd codegen) - I'll remove this paragraph.

ljk53 committed:
Update on "[pytorch][codegen] move is_abstract property to NativeFunction model"


Moved to a shared place so that gen_variable_type.py can reuse it.

Differential Revision: [D25087808](https://our.internmc.facebook.com/intern/diff/D25087808)

[ghstack-poisoned]
